### PR TITLE
pci: JSON device marshalling fixes

### PR DESCRIPTION
**Pull request by fromanirh**
PR https://github.com/jaypipes/ghw/pull/209 exposed a mashalling issue with some devices.
The issue is the quoting of " from the description of some devices (e.g. 2.5" device).

Add a test to demonstrate the bug and a possible fix.

### CODE REVIEW

**Comment by fromanirh** [comment]
I added `pci.ParseDevice` because it seems a good compromise and an acceptable addition to the API; I can add a ghw snapshot which highlights the issue and which allows us to demonstrate the bug without API changes, but it will take me some more time.

File: pkg/pci/pci.go [hunk 1]
```diff
@@ -40,69 +40,54 @@ type Device struct {
...
 
+type devIdent struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type devMarshallable struct {
+	Address   string   `json:"address"`
+	Vendor    devIdent `json:"vendor"`
+	Product   devIdent `json:"vendor"`
```
Comment (jaypipes) [inline pkg/pci/pci.go:51]: copy/paste mistake ... the json tag should be "product" :)

File: pkg/pci/pci.go [hunk 2]
```diff
@@ -40,69 +40,54 @@ type Device struct {
...
+	Subsystem devIdent `json:"vendor"`
```
Comment (jaypipes) [inline pkg/pci/pci.go:52]: and this should be "subsystem"

File: pkg/pci/pci_linux.go [hunk 3]
```diff
@@ -233,7 +238,21 @@ func (info *Info) GetDevice(address string) *Device {
 	if modaliasInfo == nil {
 		return nil
 	}
+	return info.getDeviceFromModaliasInfo(address, modaliasInfo)
+}
+
+// ParseDevice returns a pointer to a Device given its describing data.
+// The PCI device obtained this way may not exist in the system;
+// use GetDevice to get a *Device which is found in the system
+func (info *Info) ParseDevice(address, modalias string) *Device {
```
Comment (jaypipes) [inline pkg/pci/pci_linux.go:247]: technically this is introducing a public exported interface, and I realize you did this in order to unit-test it in the pci_test.go file which imports the `pkg/pci` package in order to test the exported interfaces. So, I'm OK with this.

**Review by jaypipes — CHANGES_REQUESTED** [review]
Very nice work @fromanirh! Just a little copy/pasta mistake to fix up otherwise, lgtm.

**Comment by jaypipes** [comment]
> I added `pci.ParseDevice` because it seems a good compromise and an acceptable addition to the API (ok, ok, disclosure: it's the lazy way!); I can add a ghw snapshot which highlights the issue and which allows us to demonstrate the bug without API changes, but it will take me some more time.

Yup, I'm cool with it :)

**Review by jaypipes — APPROVED** [review]
\o/ thanks @fromanirh!

**Pull request merged**